### PR TITLE
Use batch processing in edit-widgets package

### PR DESCRIPTION
**Pull request by adamziel**
This PR updates the `edit-widgets` package so that it saves the entities using batch processing against the correct API endpoints.

Testing instructions TBD

### CODE REVIEW

**Comment by TimothyBJacobs** [comment]
This looks great to me!

> Chunks processing seems to be one step too far as they make the entire transaction not atomic.

Could you expand what this means?

**Comment by adamziel** [comment]
>> Chunks processing seems to be one step too far as they make the entire transaction not atomic.
> Could you expand what this means?

@TimothyBJacobs sure! The words "transaction" and "commit" usually characterize an atomic operation that either goes through or fails in its entirety. In this PR, however, a transaction may consists of multiple chunks which means it may partially fail so I'm not entirely comfortable with this vocabulary.

I'm not saying we shouldn't do chunks - that's a useful automation. If the caller needs to save 21 entities and a single batch may only handle 20, then it's going to send two batch requests anyway. Handling that logic manually would be repetitive and prone to errors so it's good to have it baked in. And there transactions out there that may end up in a partially failed state too (MyISAM - I'm looking at you).

I think I would be more comfortable with this if the chunks were called "transactions" and transactions were called something else. This way each transaction would be atomic, but the collection of transactions wouldn't.

**Comment by adamziel** [comment]
There we go, I updated the nomenclature to "batch" and "transaction".

**Comment by TimothyBJacobs** [comment]
One issue I'm running into is that the `batch` we return to the caller takes the ordered list of items in a queue, and returns them as an object. Object insertion order isn't guaranteed. This means that if the caller doesn't have the `itemId` returned by `enqueueItem`, which in nearly all cases it won't. They can't match up their `apiFetch` calls with the results returned by `processBatch`.

Keying the `results` and `errors` by the `itemId` makes a lot of operations convenient, so maybe instead of reverting that to an array, we maintain a property on the batch of the `itemId`s in insertion order?

**Comment by youknowriad** [comment]
@adamziel would you mind explaining the public API of the proposed package in the description or the README?

File: packages/edit-widgets/src/store/batch-processing/reducer.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+						...( state.promises[ action.queue ] || {} ),
+						[ action.context ]: {
+							promise: action.promise,
+							resolve: action.resolve,
+							reject: action.reject,
+						},
+					},
+				},
+			};
+		}
```
Comment (adamziel) [inline packages/edit-widgets/src/store/batch-processing/reducer.js:107]: Promises are mutable but redux state should be immutable by definition - let's remove this in a follow up PR.

Comment (TimothyBJacobs) [reply packages/edit-widgets/src/store/batch-processing/reducer.js:107]: So putting this in just a global somewhere?

Comment (adamziel) [reply packages/edit-widgets/src/store/batch-processing/reducer.js:109]: I was thinking more about not storing this anywhere, only returning it to the caller. It's not a very well-thought proposal, I understand you added that for a reason. Let's figure it out after 9.2 RC1 - this works and is fine for now.

**Review by TimothyBJacobs — APPROVED** [review]

**Pull request merged**